labs/oas-apigee-mock/lab.md

```diff
@@ -0,0 +1,181 @@
+
+An Apigee X organisation configured for external exposure is required to
+complete this lab. See
+[here](https://github.com/apigee/devrel/tree/main/tools/apigee-x-trial-provision)
+for details on provisioning an evaluation organisation.
+
+### Tools
+
+You will need to have the following dependencies configired in you local
+enviornment in order to complete the lab tasks:
+
+- Bash (Unix shell)
+- [NodeJS](https://nodejs.org/en/) LTS version or above.
+- [gcloud](https://cloud.google.com/sdk/docs/install)
+- [git](https://git-scm.com/)
+- [apigee-sackmesser](https://github.com/apigee/devrel/tree/main/tools/apigee-sackmesser)
+- [apigeecli](https://github.com/srinandan/apigeecli)
+- [jq](https://stedolan.github.io/jq/)
+
+### Environment Variables
+
+Commands used during the lab will require the following environment variables
+to be configured.
+
+``` bash
+export TOKEN=$(gcloud auth application-default print-access-token)
+export APIGEE_ENV=<apigee-environment-name>
+export APIGEE_ORG=<gcp-project-name>
+export RUNTIME_HOST_ALIAS=<your-runtime-host-alias>
+```
+
+### oas-apigee-mock
+
+Clone the Apigee DevRel repo and install the dependencies for the
+oas-apigee-mock tool.
+
+``` bash
+git clone https://github.com/apigee/devrel.git
+cd devrel/tools/oas-apigee-mock/
+npm install
+```
+
+## Generate, deploy and test an Apigee Proxy
+
+Duration: 0:30:00
+
+### Generate an Apigee Proxy from an Open API Specification
+
+Use the `orders-apikey-header.yaml` Open API specification included
+in the `devrel/tools/oas-apigee-mock/test/oas` folder to generate a proxy bundle.
+
+``` bash
+node bin/oas-apigee-mock generateApi web-orders-proxy-v1 \
+  -s test/oas/orders-apikey-header.yaml \
+  -o
+```
+
+### Update the apickli configuration to use your Host Alias
+
+Update the `before(function ()` block in
+`test/features/step_definitions/init.js` with your organisation's hostname.
+This will be your `RUNTIME_HOST_ALIAS` if you followed the
+[Apigee X Trial Provisioning](https://github.com/apigee/devrel/tree/main/tools/apigee-x-trial-provision)
+script.
+
+``` javascript
+before(function () {
+  this.apickli = new apickli.Apickli(
+    "https",process.env.RUNTIME_HOST_ALIAS
+  );
+```
+
+### Update the test suite proxy reference
+
+Update the test suite with your proxy name, we used `web-orders-proxy-v1` in
+the generate proxy step.
+
+``` bash
+sed -i 's/oas-apigee-mock-orders-apikey-header/web-orders-proxy-v1/' test/features/orders-apikey-header.feature
+```
+
+### Test the Proxy
+
+Lets start testing by runing the test suite.
+
+``` bash
+./node_modules/.bin/cucumber-js test/features/orders-apikey-header.feature --format json:test/test_report.json --publish-quiet
+```
+
+The test should fail as we have yet to deploy the proxy we generated
+previously.
+
+### Deploy the generated proxy to your Apigee Organisation
+
+``` bash
+sackmesser deploy -d "$PWD/api_bundles/web-orders-proxy-v1" --googleapi -t "$TOKEN" -o "$APIGEE_ORG" -e "$APIGEE_ENV"
```

Comment:
Maybe replace $PWD with a variable that we set earlier like `$LAB_HOME` such that the user can move around if they want to